### PR TITLE
refactor: make ReverseSlice mutative

### PR DESCRIPTION
**Pull request by sehyunc**
Closes: #2181 

## What is the purpose of the change

Refactor `ReverseSlice` utility function to be mutative and reverse in place.

## Brief Changelog

* refactor existing `ReverseSlice` function
* add tests in new test file specifically for this function

## Testing and Verifying

This change added tests and can be verified as follows:
* `TestReverseSlice` in `slice_helper_test.go`

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not documented)

### CODE REVIEW

**Review by stackman27 — COMMENTED** [review]
Left a bit of the comments. Congratulations on the PR!! 😎

File: osmoutils/slice_helper_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		"Empty array":             {[]string{}, []string{}},
+		"Odd length array":        {[]string{"a", "b", "c"}, []string{"c", "b", "a"}},
+		"Single element array":    {[]string{"a"}, []string{"a"}},
+		"Array with empty string": {[]string{"a", "b", "c", "", "d"}, []string{"d", "", "c", "b", "a"}},
+		"Array with numbers":      {[]string{"a", "b", "c", "1", "2", "3"}, []string{"3", "2", "1", "c", "b", "a"}},
+	}
+
+	for _, tc := range tests {
+		actualSolvedInput := ReverseSlice(tc.s)
+		require.True(t, reflect.DeepEqual(actualSolvedInput, tc.expectedSolvedInput))
```
Comment (stackman27) [inline osmoutils/slice_helper_test.go:26]: Not sure if `require.True` is necessary while using DeepEqual 

```suggestion
		reflect.DeepEqual(actualSolvedInput, tc.expectedSolvedInput)
```

File: osmoutils/slice_helper.go [hunk 2]
```diff
@@ -24,13 +24,14 @@ func Filter[T interface{}](filter func(T) bool, s []T) []T {
 	return filteredSlice
 }
 
-// ReverseSlice returns a reversed copy of the input slice.
-// Does not mutate argument.
+// Reverse slice reverses the input slice in place.
```
Comment (stackman27) [inline osmoutils/slice_helper.go:27]:
```suggestion
// ReverseSlice reverses the input slice in place.
```

File: osmoutils/slice_helper_test.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	}{
+		"Even length array":       {[]string{"a", "b", "c", "d"}, []string{"d", "c", "b", "a"}},
+		"Empty array":             {[]string{}, []string{}},
+		"Odd length array":        {[]string{"a", "b", "c"}, []string{"c", "b", "a"}},
+		"Single element array":    {[]string{"a"}, []string{"a"}},
+		"Array with empty string": {[]string{"a", "b", "c", "", "d"}, []string{"d", "", "c", "b", "a"}},
+		"Array with numbers":      {[]string{"a", "b", "c", "1", "2", "3"}, []string{"3", "2", "1", "c", "b", "a"}},
+	}
+
+	for _, tc := range tests {
```
Comment (stackman27) [inline osmoutils/slice_helper_test.go:24]: We can probably wrap this up with something like; 

```
for name, tc := range tests {
t.Run(name, func(t *testing.T) {
 ...
}
}
```

File: osmoutils/slice_helper_test.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestReverseSlice(t *testing.T) {
+	tests := map[string]struct {
+		s []string
+
+		expectedSolvedInput []string
+	}{
+		"Even length array":       {[]string{"a", "b", "c", "d"}, []string{"d", "c", "b", "a"}},
```
Comment (AlpinYukseloglu) [inline osmoutils/slice_helper_test.go:16]:
```suggestion
		"Even length array":       {
		    s: []string{"a", "b", "c", "d"}, 
		    expectedSolvedInput: []string{"d", "c", "b", "a"}
		},
```

**Review by AlpinYukseloglu — COMMENTED** [review]
Thanks for this PR! Other than the comment I left about test structure/readability, the core logic LGTM! It would be great if you added error case(s) just to be comprehensive though imo

File: osmoutils/slice_helper_test.go (hunk 1 above)
Comment (p0mvn) [reply osmoutils/slice_helper_test.go:26]: `DeepEqual` returns a boolean but does not assert. We still would like to make the assertion on the result

Comment (p0mvn) [reply osmoutils/slice_helper_test.go:26]: Suggest trying the following if the goal is to simplify:
```
require.EqualValues(t, tc.expectedSolvedInput, actualSolvedInput)
```

File: osmoutils/slice_helper_test.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package osmoutils
```
Comment (p0mvn) [inline osmoutils/slice_helper_test.go:1]:
```suggestion
package osmoutils_test
```

**Review by p0mvn — COMMENTED** [review]
Nice work!

File: osmoutils/slice_helper_test.go (hunk 1 above)
Comment (ValarDragon) [reply osmoutils/slice_helper_test.go:26]: You can also just do `require.Equal` for slices,. https://play.golang.com/p/HKavaKV9bd0

**Review by ValarDragon — APPROVED** [review]
LGTM after minor notes added by others

File: osmoutils/slice_helper_test.go (hunk 5 above)
Comment (sehyunc) [reply osmoutils/slice_helper_test.go:1]: When I make this change, I get the `undeclared name: ReverseSlice` error. Looking at `binary_search_test.go`, it also has package set to `osmoutils`. Have I configured something incorrectly?

Comment (stackman27) [reply osmoutils/slice_helper_test.go:1]: You can fix this by importing osmoutils like below. I'm not sure why we dont use package osmoutils_test in `binary_search_test.go`. Maybe it's worth changing both or keeping this package name the same as `binary_search_test.go` 

```
import (
	"github.com/osmosis-labs/osmosis/v10/osmoutils"
...
)

osmoutils.ReverseSlice(...)
```

Comment (sehyunc) [reply osmoutils/slice_helper_test.go]: Thanks for the tip sir. Still getting used to Go 😅

**Comment by ValarDragon** [comment]
Thanks for the PR!

**Pull request merged**